apps/wallet-mobile/src/features/Swap/common/helpers.ts

```diff
@@ -95,3 +96,26 @@ export const sortTokensByName = (a: Balance.TokenInfo, b: Balance.TokenInfo, wal
 
   return nameA.localeCompare(nameB, undefined, {sensitivity: 'base'})
 }
+
+export const calculatePriceImpactRisk = (priceImpact: number) => {
+  if (priceImpact >= 1 && priceImpact <= 10) {
+    return 'warning'
+  }
+  if (priceImpact > 10) {
+    return 'negative'
+  }
+  return 'positive'
+}
+
+export const priceImpactColorObject = (theme: Theme) => {
+  return {
+    warning: theme.color.yellow[500],
+    negative: theme.color.magenta[500],
+  }
+}
+export const priceImpactBannerColorObject = (theme: Theme) => {
+  return {
+    warning: theme.color.yellow[100],
+    negative: theme.color.magenta[100],
+  }
+}
```

Comment:
export it out to a hook `usePriceImpactTheme` 
```suggestion
export const usePriceImpactStatusTheme = (status: SwapPriceImpactStatus) {
  const theme = useTheme()
  
  if (status === 'negative') {
    return {
      text: theme.color.magenta[500],
      background: theme.color.magenta[100]
    }
  } else if (status === 'warning') {
    return {
      text: theme.color.yellow[500],
      background: theme.color.yellow[100]
    }
  }
  
  return {
    // TODO: fix
    text: theme.color.yellow[500],
    background: theme.color.yellow[100]
  }
}
```